src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -181,7 +189,8 @@ public DependencyContext Build()
                 compilationLibraries = compilationLibraries
                     .Concat(GetReferenceAssemblyLibraries())
                     .Concat(GetLibraries(compilationExports, libraryLookup, dependencyLookup, runtime: false).Cast<CompilationLibrary>())
-                    .Concat(GetDirectReferenceCompilationLibraries());
+                    .Concat(GetDirectReferenceCompilationLibraries())
+                    .Concat(GetDependencyReferenceCompilationLibraries());
```

Comment:
I thought you mentioned that these don't show up as references to the compiler.  If they don't show up as references to the compiler, then I don't think they should show up as `compilationLibraries` here either.

Reply:
Right. I'll fix this up.